datafusion/core/tests/sqllogictests/test_files/timestamps.slt

```diff
@@ -392,11 +392,28 @@ drop table ts_data_millis
 statement ok
 drop table ts_data_secs
 
-
-
 ##########
 ## test date_bin function
 ##########
+
+# not support interval 0
+statement error Execution error: DATE_BIN stride must be non-zero and positive
+SELECT DATE_BIN(INTERVAL '0 second', TIMESTAMP '2022-08-03 14:38:50.000000006Z', TIMESTAMP '1970-01-01T00:00:00Z')
+
+statement error Execution error: DATE_BIN stride must be non-zero and positive
+SELECT DATE_BIN(INTERVAL '0 month', TIMESTAMP '2022-08-03 14:38:50.000000006Z')
+
+statement error Execution error: DATE_BIN stride must be non-zero and positive
+SELECT
+  DATE_BIN(INTERVAL '0' minute, time) AS time,
+  count(val)
+FROM (
+  VALUES
+    (TIMESTAMP '2021-06-10 17:05:00Z', 0.5),
+    (TIMESTAMP '2021-06-10 17:19:10Z', 0.3)
+  ) as t (time, val)
+group by time;
```

Comment:
I remember @alamb want just right number of tests, I think these 3 tests are good to capture 2 different forms of stride on constant (second --> nanosecond in the code, and month ->  month) and an aggregate on data.